src/participant.rs

```diff
@@ -285,38 +288,26 @@ pub(crate) trait InnerProtocolParticipant: ProtocolParticipant {
 
     /// Process a `ready` message: tell other participants that we're ready and
     /// see if all others have also reported that they are ready.
-    fn process_ready_message<T: TypeTag<Value = ()>>(
+    fn process_ready_message<R: RngCore + CryptoRng>(
         &mut self,
+        rng: &mut R,
         message: &Message,
     ) -> Result<(ProcessOutcome<Self::Output>, bool)> {
-        self.local_storage_mut().store::<T>(message.from(), ());
-        let empty: [u8; 0] = [];
-        // If message came from self, then tell the other participants that we are ready
-        let self_initiated_outcome = if message.from() == self.id() {
-            let messages = self
-                .other_ids()
+        // If message came from self, then we should start the protocol
+        if message.from() == self.id() {
+            // First, process any messages that had been received before the Ready signal
+            let banked_messages = self.fetch_all_messages()?;
+            self.set_ready();
+            let outcomes = banked_messages
                 .iter()
-                .map(|other_id| {
-                    Message::new(
-                        message.message_type(),
-                        message.id(),
-                        self.id(),
-                        *other_id,
-                        &empty,
-                    )
-                })
-                .collect::<Result<Vec<Message>>>()?;
-            ProcessOutcome::Processed(messages)
-        } else {
-            ProcessOutcome::Incomplete
-        };
+                .map(|m| self.process_message(rng, m))
+                .collect::<Result<Vec<ProcessOutcome<Self::Output>>>>()?;
 
-        // Make sure that all parties are ready before proceeding
-        let is_ready = self
-            .local_storage()
-            .contains_for_all_ids::<T>(&self.all_participants());
-
-        Ok((self_initiated_outcome, is_ready))
+            Ok((ProcessOutcome::collect(outcomes)?, true))
```

Comment:
todo: Since this always returns true, we should remove the bool from the return type and remove the false branch from any conditional that depends on it.